### DISCUSSION TITLE
Attribute AndroidApp gets added to entry

### DISCUSSION THREAD

**Discussion by ghost** [Kunzisoft/KeePassDX#1340] [General]
I've noticed that under some circumstances the attribute AndroidApp is added to an entry. This makes sense to store credentials for an app. However, this is also added when using a browser. When switching to Magic Keyboard, KeePassDX automatically loads this entry regardless which URL is actually opened in the browser. If you're not paying full attention you might send the wrong credentials to the wrong website.

Of course, I want to use KeePassDX in a browser. That's probably the regular case anyway. So adding the browser(s) to the app blocklist wouldn't make sense.

Can I prevent this somehow? Is this a bug or is a new feature required?

**Comment by J-Jamet** [comment]
When you click on the "key" icon on the keyboard, the specified package name will be searched, and if saving is enabled, the entry will simply save the package in the AndroidApp field.
If you want the URL, you have to share it from your browser.
If you don't want to search any information during the selection mode, you have to do a long click.
All this is already described in the wiki : https://github.com/Kunzisoft/KeePassDX/wiki/Magikeyboard#usage

**Reply by ghost** [reply]
Thanks for your response. Obviously, I missed this information in the wiki or didn't read it carefully enough. So I guess, allowing the user to blocklist app i.e. browser from getting added as AndroidApp is not an option.

Part of the reason for using Magic Keyboard at all is this issue:
https://github.com/Kunzisoft/KeePassDX/issues/1272

**Reply by J-Jamet** [reply]
It could be a new feature to create, but I don't see the point since by going through the Magikeyboard, the recognition is not automatic and it is up to the user to choose the way to send the information for the selection.

**Reply by J-Jamet** [reply]
I understand that the autofill is to be improved but I don't want to put information of some forms directly in the code (by directly coding the URL and the formatting), it would favor some services more than others and it's not modular. For me the solution is to create a specific configuration per form to detect the autofill fields (https://github.com/Kunzisoft/KeePassDX/issues/1272#issuecomment-1086947929).

**Reply by ghost** [reply]
Knowing nothing about the autofill api, I wonder if KeePassDX couldn't simply offer to select an entry AND the attribute manually when you request Autofill for a random input field. This would effectively work like Magic Keyboard without the side effects.

**Reply by J-Jamet** [reply]
This is not possible, it is the service of the device that manages the autofill events. Otherwise it implies to display the autofill in each field of all forms... This is not really a desired behavior.

**Reply by ghost** [reply]
Would you consider adding an option "Exclude Browser app from AndroidApp"? The ID of the default Browser app should be available through some Android API, I hope, and it wouldn't require to maintain a list of Android app IDs. According to this https://stackoverflow.com/a/48616636 there's a way to list all browser apps. I would even prefer this, as I use up to three browsers for reasons.

**Reply by nomis** [reply]
Keepass2Android avoids this problem by asking the user each time if they want to save the app or not. That would let you avoid associating the web browser with an entry.
